macros/schema_tests/equality.sql

```diff
@@ -18,8 +25,15 @@ information schema — this allows the model to be an ephemeral model
 {%- endif -%}
 
 {% set compare_model = kwargs.get('compare_model', kwargs.get('arg')) %}
-{% set compare_columns = kwargs.get('compare_columns', adapter.get_columns_in_relation(model) | map(attribute='quoted') ) %}
-{% set compare_cols_csv = compare_columns | join(', ') %}
+{% set compare_cols_a = kwargs.get('compare_columns', adapter.get_columns_in_relation(model) | map(attribute='quoted') ) %}
+{% set compare_cols_a_csv = compare_cols_a | join(', ') %}
+
+{%- if kwargs.get('all_columns_present_in_both_tables', false) -%}
+    {% set compare_cols_b = kwargs.get('compare_columns', adapter.get_columns_in_relation(compare_model) | map(attribute='quoted') ) %}
+    {% set compare_cols_b_csv = compare_cols_b | join(', ') %}
+{%- else -%}
+    {% set compare_cols_b_csv = compare_cols_a_csv %}
```

Comment:
So this will fail if the columns are in a different order, right?:
| id | name   | 
|----|--------| 
| 1  | claire | 

| name | id |
|--|--|
| claire | 1 |

I'd  posit that these two tables are in fact equal though — curious for your thoughts here!

Reply:
Yes order would cause the test to fail, but I would agree those two are equal in content. I think rather than just the `all_columns_present_in_both_tables` variable having two would be more appropriate:
```
identical_columns: true|false
identical_column_order: true|false
```

The case I currently want to use this test for is a query that must match a report format. So I want to test both content and column order. Appreciate not everyone would care about the column order though.

Reply:
I spent way too long thinking about this this morning

I think if we were implementing the macro from scratch, we'd do something to be smarter about how the choose which columns to compare, and have something like:
```
comparison_type is one of ['all_columns_ordered', 'all_columns_unordered', 'columns_present_in_both', 'user_specifed']
```
Alas, as it stands, it's a little hard to change the default behavior of a utils macro, as it will cause breaking changes.

I think your proposal is a good one, and adding `identical_columns` and `identical_column_order` args with defaults of false (so as not to introduce breaking changes) is the way to go!